### PR TITLE
Release automations

### LINKED ISSUE #7: Auto-release on PyPI and auto-changelog
OTOH this seems to be so GitHub-specific, and if we ever move to something else away from MSFT ([Gitea](https://gitea.io)?) we nearly lose all this.

### PR DESCRIPTION
**Pull request by knopki**
Closes #7 

Refactor workflows:

PR (before merge): Lint, test, build.

Push to `master`:
Test, maybe bump version (conventional commits specification).
If no version bump - just build.
If version bump - create git tag, bump version in files, generate and commit changelog, build, create draft github release.

On release publish: Build and upload to PyPI.

@blokhin Please, set `PYPI_API_TOKEN` secret.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ build-backend = "flit_core.buildapi"
 
 [project]
 name = "metis-client"
-dynamic = ["version"]  # read from metis_client/__init__.py
+version = "0.0.3"
```
Comment (blokhin) [inline pyproject.toml:8]: @knopki no more read from `metis_client/__init__.py` ?

**Review by blokhin — APPROVED** [review]

**Comment by blokhin** [comment]
@knopki yep, done :hand:

File: pyproject.toml (hunk 1 above)
Comment (knopki) [reply pyproject.toml:8]: `commitizen` should update this on bump, so dynamic import is useless.

**Comment by knopki** [comment]
@blokhin Please, allow `commitizen-tools/commitizen-action`, `softprops/action-gh-release` and `pypa/gh-action-pypi-publish` workflows.

**Comment by blokhin** [comment]
@knopki done :heavy_check_mark:

**Pull request merged**